File: 0389-find-the-difference/0389-find-the-difference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        sorted_s = sorted(s)
        sorted_t = sorted(t)
        
        if not sorted_s:  # Check if sorted_s is empty
            return sorted_t[-1]
        
        for i in range(len(sorted_s)):  # Iterate over indices of sorted_s
            if sorted_t[i] != sorted_s[i]:
                return sorted_t[i]
        
        # If no difference found yet, the extra character is the last character of t
        return sorted_t[-1]
            

File: 0389-find-the-difference/0389-find-the-difference.java
class Solution {
    public char findTheDifference(String s, String t) {
        char[] sortedS = s.toCharArray();
        char[] sortedT = t.toCharArray();
        
        Arrays.sort(sortedS);
        Arrays.sort(sortedT);
        
        if (sortedS.length == 0) { // Check if sortedS is empty
            return sortedT[sortedT.length - 1];
        }
        
        for (int i = 0; i < sortedS.length; i++) { // Iterate over indices of sortedS
            if (sortedT[i] != sortedS[i]) {
                return sortedT[i];
            }
        }
        
        // If no difference found yet, the extra character is the last character of t
        return sortedT[sortedT.length - 1];
    }
}